serveradmin/servershell/views.py

```diff
@@ -156,12 +198,17 @@ def get_results(request):
         restrict = shown_attributes.copy()
         if 'servertype' not in restrict:
             restrict.append('servertype')
-        query = Query(parse_query(term), restrict, order_by)
+        main_query = Query(parse_query(term), restrict, order_by)
+
+        query = MergedQuery([
+            Query({'object_id': Any(*pinned)}, restrict, None),
```

Comment:
You can ommit the third parameter if not used.
```suggestion
            Query({'object_id': Any(*pinned)}, restrict),
```